### ISSUE TITLE
code coverage upload fails with Error in readRDS(files[i]) : error reading from connection

### ISSUE THREAD

**Issue by ajabadi** [r-lib/covr#451]
### Observed behaviour

Uploading the code coverage succeeds on the fork but fails on main repo (same commit) with:

> Run covr::codecov(type = c('tests'))
> Error: Error in readRDS(files[i]) : error reading from connection
> Calls: <Anonymous> -> package_coverage -> merge_coverage
> Execution halted
> Error: Process completed with exit code 1.

Both used macOS-latest. Both public and no token used. Both repos added on codecov.io.

### Expected behaviour

When both the fork and main repo are (public and) added to codecov.io, we should get separate coverage reports for a push to both (while under the hood they could be the same report)

the fork Action: https://github.com/ajabadi/mixOmics/runs/1174435842
main repo action: https://github.com/mixOmicsTeam/mixOmics/runs/1174532979

the fork on codecov: https://codecov.io/gh/ajabadi/mixOmics/commits
main repo on codecov: https://codecov.io/gh/mixOmicsTeam/mixOmics

**Comment by ajabadi** [comment]
Update: It seems to behave randomly on re-runs. Sometimes running successfully while other times failing.

**Comment by jimhester** [comment]
covr writes the coverage information when the R process (or any child R process) exits, via a exit finalizer using `reg.finalizer(onexit = TRUE)`

I think likely something is killing the R process before the coverage information can be fully written, so the Rds files are being corrupted. Sometimes the data finishes writing before getting interrupted and it works, but other times it does not and fails.

I would look through your code and its dependencies for timeouts or `SIGKILL` signals and make sure they are set high enough for the data to be written fully and preferably use `SIGINT` so the process can exit cleanly.

**Comment by ajabadi** [comment]
Thanks @jimhester 

I had followed the example in https://github.com/r-lib/actions/tree/master/examples#test-coverage-workflow
For some reason, it happens on `macOS-latest` but not on `ubuntu-latest` (on which it runs just fine so I use the latter to upload the coverage files). 

Also in later runs, it was not just an issue on the main repo but also on the fork. I'm not sure but my wild guess would be that it was caused by some recent changes to `macOS-latest`.
